packages/web3-common/src/json_rpc.ts

```diff
@@ -33,13 +34,22 @@ export const isResponseWithError = <Error = unknown, Result = unknown>(
 	(typeof response.id === 'number' || typeof response.id === 'string');
 
 export const isResponseWithNotification = <Result>(
-	response: JsonRpcNotification<Result>,
+	response: JsonRpcNotification<Result> | JsonRpcSubscriptionResult,
```

Comment:
we should only allow passing `JsonRpcNotification` type here.

Reply:
The response we get is a mix of these two types.